pkg/packageinstall/packageinstall.go

```diff
@@ -266,3 +278,62 @@ func (pi *PackageInstallCR) update(updateFunc func(*pkgingv1alpha1.PackageInstal
 
 	return fmt.Errorf("Updating package install: %s", lastErr)
 }
+
+func (pi *PackageInstallCR) createPlaceHolderSecrets(pv *datapkgingv1alpha1.Package) error {
+	for i, fetch := range pv.Spec.Template.Spec.Fetch {
```

Comment:
instead of modifying Package, lets make a DeepCopy() and return a modified copy.